src/common/time/src/interval.rs

```diff
@@ -0,0 +1,755 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::default::Default;
+use std::fmt::{self, Display, Formatter, Write};
+use std::hash::{Hash, Hasher};
+
+use arrow::datatypes::IntervalUnit as ArrowIntervalUnit;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::error::ParseIntervalSnafu;
+
+#[derive(
+    Debug, Default, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize,
+)]
+pub enum IntervalUnit {
+    /// Indicates the number of elapsed whole months, stored as 4-byte integers.
+    YearMonth,
+    /// Indicates the number of elapsed days and milliseconds,
+    /// stored as 2 contiguous 32-bit integers (days, milliseconds) (8-bytes in total).
+    DayTime,
+    /// A triple of the number of elapsed months, days, and nanoseconds.
+    /// The values are stored contiguously in 16 byte blocks. Months and
+    /// days are encoded as 32 bit integers and nanoseconds is encoded as a
+    /// 64 bit integer. All integers are signed. Each field is independent
+    /// (e.g. there is no constraint that nanoseconds have the same sign
+    /// as days or that the quantity of nanoseconds represents less
+    /// than a day's worth of time).
+    #[default]
+    MonthDayNano,
+}
+
+impl From<&ArrowIntervalUnit> for IntervalUnit {
+    fn from(unit: &ArrowIntervalUnit) -> Self {
+        match unit {
+            ArrowIntervalUnit::YearMonth => IntervalUnit::YearMonth,
+            ArrowIntervalUnit::DayTime => IntervalUnit::DayTime,
+            ArrowIntervalUnit::MonthDayNano => IntervalUnit::MonthDayNano,
+        }
+    }
+}
+
+impl From<ArrowIntervalUnit> for IntervalUnit {
+    fn from(unit: ArrowIntervalUnit) -> Self {
+        (&unit).into()
+    }
+}
+
+/// Interval data format
+/// | months | days   | micros     |
+/// | 4bytes | 4bytes | 8bytes     |  
+/// All interval types are stored in this format.
+#[derive(Debug, Clone, Default, Copy, Serialize, Deserialize)]
+pub struct Interval {
```

Comment:
The current interval implementation uses microsecond precision,`YearMonth/DayTime/MonthDayNano` will convert to this `interval` type which may be confusing.for `MonthDayNano` it would lose the precision of the last three digits, if we encounter a situation like the doc(https://arrow.apache.org/docs/python/generated/pyarrow.month_day_nano_interval.html#pyarrow.month_day_nano_interval). 
Additionally, I calculated that using `i64` to represent nanoseconds would give us a range of approximately 292 years, whereas using microseconds would give us a range of approximately 292,471 years. it's unlikely that we would need such a large range for interval data (and precision may be more important to our scenarios ). And I will modify the `microsecond` to `nanosecond` in `interval` later.

Reply:
> The current interval implementation uses microsecond precision,`YearMonth/DayTime/MonthDayNano` will convert to this `interval` type which may be confusing.for `MonthDayNano` it would lose the precision of the last three digits, if we encounter a situation like the doc(https://arrow.apache.org/docs/python/generated/pyarrow.month_day_nano_interval.html#pyarrow.month_day_nano_interval). Additionally, I calculated that using `i64` to represent nanoseconds would give us a range of approximately 292 years, whereas using microseconds would give us a range of approximately 292,471 years. it's unlikely that we would need such a large range for interval data (and precision may be more important to our scenarios ). And I will modify the `microsecond` to `nanosecond` in `interval` later.

I also recommend change `micros` to `nanos` to avoid confusion.